src/modules/75-cf/pages/feature-flags-detail/flag-pipeline-tab/__tests__/FlagPipelineTab.test.tsx

```diff
@@ -146,9 +174,184 @@ describe('FlagPipelineTab', () => {
       })
       userEvent.click(addFlagPipelineButton)
 
+      await waitFor(() => expect(screen.getByText('Loading, please wait...')).toBeInTheDocument())
+    })
+
+    test.each([undefined, null, { availablePipelines: [] }])(
+      'it should show correct message when availablePipelines object is %s',
+      async data => {
+        jest.spyOn(cfServicesMock, 'useGetAvailableFeaturePipelines').mockReturnValue({
+          data,
+          loading: false,
+          refetch: jest.fn()
+        } as any)
+
+        renderComponent()
+        const addFlagPipelineButton = screen.getByRole('button', {
+          name: 'cf.featureFlags.flagPipeline.noDataButtonText'
+        })
+        userEvent.click(addFlagPipelineButton)
+
+        await waitFor(() =>
+          expect(screen.getByText('cf.featureFlags.flagPipeline.noAvailablePipelines')).toBeInTheDocument()
+        )
+      }
+    )
+  })
+
+  describe('Flag pipeline configured', () => {
+    const getFeaturePipelineRefetchMock = jest.fn()
+    beforeEach(() => {
+      jest.spyOn(cfServicesMock, 'useGetFeaturePipeline').mockReturnValue({
+        data: {
+          executionHistory: [],
+          pipelineConfigured: true,
+          pipelineDetails: {
+            identifier: 'pipeline5',
+            name: 'Pipeline 5',
+            description: 'This is a test pipeline'
+          }
+        },
+        loading: false,
+        refetch: getFeaturePipelineRefetchMock
+      } as any)
+    })
+    test('it should update flag pipeline correctly', async () => {
```

Comment:
A wee bit of whitespace makes it easier to visually scan a file
```suggestion
    })
    
    test('it should update flag pipeline correctly', async () => {
```